benchmarks/protoboeuf/benchmark_pb.rb

```diff
@@ -0,0 +1,4888 @@
+# frozen_string_literal: true
+module ProtoBoeuf
+class Position
+  def self.decode(buff)
+    buff = buff.dup
+    buff.force_encoding("UTF-8")
+    allocate.decode_from(buff, 0, buff.bytesize)
+  end
+
+  # required field readers
+  attr_accessor :x, :y, :z
+
+  # enum readers
+
+  # enum writers
+
+  def initialize(x: 0, y: 0, z: 0)
+    @x = x
+    @y = y
+    @z = z
+  end
+
+  def decode_from(buff, index, len)
+    
+      @x = 0
+      @y = 0
+      @z = 0
+
+            tag = buff.getbyte(index)
+        index += 1
+
+
+    while true
+      if tag == 0x8
+                ## PULL_UINT64
+        @x =       if (byte0 = buff.getbyte(index)) < 0x80
```

Comment:
Is this file generated? The indentation etc looks quite odd.
This looks like manual inlining, is it that beneficial with YJIT to be worth it?

Reply:
Yes